modules/afsocket/afsocket-source.c

```diff
@@ -679,6 +679,27 @@ afsocket_sd_setup_reader_options(AFSocketSourceDriver *self)
       if (self->dynamic_window_size != 0)
         min_iw_size_per_reader = 1;
 
+      if (self->max_connections > 0 && self->dynamic_window_size > 0)
```

Comment:
As this function is still not a small one, I'd recommend to factor this change to a separate function.
Name of this new function could be `_correct_dynamic_window_size_if_needed`, or, `_adjust_dynamic_window_size` .